ex9/js/main.js

```diff
@@ -0,0 +1,38 @@
+requirejs(['ajaxGet','load','filter', 'sort'] , function(ajaxGet, load, filter, sort) {
+  $(function() {
+    var phoneBookData = {};
+    var render = function(objList) {
+      $('li').remove();
+      $("#phoneBookTemplate").tmpl( objList )
+      .appendTo( "#phoneBook" );
+
+    // Remove li element from list
+    $("li").on("click",function(elementID) {
+      console.log('testing',elementID);
+      $(this).remove();
+    });
+
+
+    };
+
+    load(function() {
+      ajaxGet('json/data.json', function(phoneBookDataA) {
+        phoneBookData = phoneBookDataA;
+        render( phoneBookData );
+      })
+    });
+
+    filter(function() {
+      var filterList = _.filter(phoneBookData, function(name) {
+        if (name.name.indexOf($( '#filter' ).val()) > -1) return true;
```

Comment:
Instead of here accessing the "internal" of the filter UI, have the caller (in filter.js) pass the filter value into this the function filter calls back (line 25 above).